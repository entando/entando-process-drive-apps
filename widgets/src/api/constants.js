export const DOMAINS = {
  PDA: process.env.REACT_APP_DOMAIN || '',
  APP_BUILDER: process.env.REACT_APP_APP_BUILDER_DOMAIN || '',
};

export const IS_MOCKED_API = process.env.REACT_APP_MOCKED_API === 'true';

export const MOCK_API_DELAY = 800;

export const LOCAL = process.env.REACT_APP_LOCAL === 'true';

export const METHODS = {
  GET: 'GET',
  POST: 'POST',
  PUT: 'PUT',
  DELETE: 'DELETE',
};

// Inject ur access_token if you are trying to reach the APP-BUILDER API locally
export const TOKENS = {
  APP_BUILDER: '0694821a5bdc2229aac820e8a773669a',
};
