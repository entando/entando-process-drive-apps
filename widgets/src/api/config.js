import { DOMAIN, IS_MOCKED_API, MOCK_API_DELAY } from 'api/constants';
import utils from 'utils';
import config from 'mocks/config';

const get = async widgetCode => {
  if (IS_MOCKED_API) {
    await utils.timeout(MOCK_API_DELAY);
    return config;
  }

  const url = `${DOMAIN}/configs/${widgetCode}`;
  const response = await fetch(url);

  return response.json();
};

export default {
  get,
};
