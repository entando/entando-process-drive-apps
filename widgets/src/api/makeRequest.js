import { IS_MOCKED_API, MOCK_API_DELAY, DOMAINS, METHODS } from 'api/constants';
import utils from 'utils';

const getParams = queryParams =>
  Object.keys(queryParams)
    .filter(k => queryParams[k] !== undefined && queryParams[k] !== '')
    .map(k => `${encodeURIComponent(k)}=${encodeURIComponent(queryParams[k])}`)
    .join('&');

export default async ({
  domain,
  uri,
  method,
  mockResponse = {},
  useAuthentication,
  body,
  headers = {},
  queryParams,
  forceMock,
}) => {
  if (IS_MOCKED_API || forceMock) {
    await utils.timeout(MOCK_API_DELAY);
    return mockResponse;
  }

  const url = `${domain || DOMAINS.PDA}${uri}${queryParams ? `?${getParams(queryParams)}` : ''}`;

  const requestHeaders = useAuthentication
    ? new Headers({
        'Content-Type': 'application/json',
        Authorization: `Bearer ${localStorage.getItem('token')}`,
        ...headers,
      })
    : new Headers(headers);

  const configs = {
    method: method || METHODS.GET,
    ...(body ? { body } : {}),
    headers: requestHeaders,
  };

  const response = await fetch(url, configs);
  if (!response.ok) {
    const payload = await response.json();
    throw new Error(
      payload.message
        ? payload.message
        : `Bad Response from server: ${response.status} ${response.statusText}`
    );
  }

  const responseHeaders = response.headers.get('Content-Type');

  if (responseHeaders && responseHeaders.includes('xml')) {
    return response.text();
  }
  if (responseHeaders && responseHeaders.includes('octet')) {
    return response;
  }
  return response.json();
};
