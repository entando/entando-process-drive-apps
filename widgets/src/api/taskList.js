import { IS_MOCKED_API, MOCK_API_DELAY, SERVICE } from 'api/constants';
import getTasks from 'mocks/taskList/api.mock';
import utils from 'utils';

const get = async (connection, page, pageSize, sortedColumn, sortOrder, filter) => {
  if (IS_MOCKED_API) {
    await utils.timeout(MOCK_API_DELAY);
    return getTasks(page, pageSize, sortedColumn, sortOrder, filter);
  }

  const url = `${SERVICE.URL}/connections/${connection}/tasks?sort=taskId`;
  const token = localStorage.getItem('token');
  const response = await fetch(url, {
    method: 'get',
    headers: new Headers({
      Authorization: `Bearer ${token}`,
      'Content-Type': 'application/json',
    }),
  });

  return response.json();
};

export default {
  get,
};
