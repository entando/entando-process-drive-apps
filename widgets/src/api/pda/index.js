import makeRequest from 'api/makeRequest';
import { SERVICE, METHODS } from 'api/constants';

import MOCK_CONNECTIONS from 'mocks/pda/connections';
import MOCK_PROCESSES from 'mocks/pda/processes';

export const getConnections = async () =>
  makeRequest({
    domain: SERVICE.URL,
    uri: '/connections',
    method: METHODS.GET,
    mockResponse: MOCK_CONNECTIONS,
    useAuthentication: false,
  });

export const getProcess = async connection =>
  makeRequest({
    domain: SERVICE.URL,
    uri: `/connections/${connection}/processes/definitions`,
    method: METHODS.GET,
    mockResponse: MOCK_PROCESSES,
    useAuthentication: false,
  });
