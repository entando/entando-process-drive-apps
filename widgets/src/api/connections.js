import { METHODS, DOMAINS } from 'api/constants';

import { CONNECTIONS } from 'mocks/taskList/configs';
import makeRequest from 'api/makeRequest';

// eslint-disable-next-line import/prefer-default-export
export const getConnections = async () =>
  makeRequest({
    domain: DOMAINS.PDA,
    uri: '/connections',
    method: METHODS.GET,
    mockResponse: CONNECTIONS,
    useAuthentication: false,
  });
