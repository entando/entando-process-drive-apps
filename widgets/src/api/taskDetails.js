import { SERVICE, METHODS } from 'api/constants';
import mockedTask from 'mocks/taskDetails/task';

import { CONNECTIONS, PROCESS, GROUPS, COLUMNS } from 'mocks/taskList/configs';
import makeRequest from 'api/makeRequest';

export const getTask = async (connection, taskId) => {
  return makeRequest({
    domain: SERVICE.URL,
    uri: `/connections/${connection}/tasks/${taskId}`,
    method: METHODS.GET,
    mockResponse: mockedTask,
    useAuthentication: true,
  });
};

// Configs

export const getConnections = async () =>
  makeRequest({
    domain: SERVICE.URL,
    uri: '/connections',
    method: METHODS.GET,
    mockResponse: CONNECTIONS,
    useAuthentication: false,
  });

export const getProcess = async connection =>
  makeRequest({
    // domain: PDA_DOMAIN,
    uri: `/connections/${connection}/processes/definitions`,
    method: METHODS.GET,
    mockResponse: PROCESS,
    useAuthentication: true,
  });

export const getGroups = async connection =>
  makeRequest({
    // domain: PDA_DOMAIN,
    uri: `/connections/${connection}/groups`,
    method: METHODS.GET,
    mockResponse: GROUPS,
    useAuthentication: true,
  });

export const getColumns = async connection =>
  makeRequest({
    // domain: PDA_DOMAIN,
    uri: `/connections/${connection}/tasks/columns`,
    method: METHODS.GET,
    mockResponse: COLUMNS,
    useAuthentication: true,
  });
