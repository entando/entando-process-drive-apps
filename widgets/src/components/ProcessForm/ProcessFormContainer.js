import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';

import theme from 'theme';
import CustomEventContext from 'components/common/CustomEventContext';
import WidgetBox from 'components/common/WidgetBox';
import JSONForm from 'components/common/form/JSONForm';
import ErrorNotification from 'components/common/ErrorNotification';
import { getProcessForm, postProcessForm } from 'api/pda/processes';
import { getPageWidget } from 'api/app-builder/pages';

class ProcessFormContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      config: null,
      loading: false,
      submitting: false,
      formSchema: null,
      errorMessage: '',
    };

    this.closeNotification = this.closeNotification.bind(this);
    this.handleError = this.handleError.bind(this);
    this.fetchSchema = this.fetchSchema.bind(this);
    this.submitProcessForm = this.submitProcessForm.bind(this);
  }

  componentDidMount() {
    this.setState({ loading: true }, async () => {
      const config = await this.fetchWidgetConfigs();

      this.setState({ config }, async () => {
        const formSchema = await this.fetchSchema();
        this.setState({ formSchema, loading: false });
      });
    });
  }

  closeNotification = () => {
    this.setState({ errorMessage: '' });
  };

  async fetchWidgetConfigs() {
    const { pageCode, frameId } = this.props;
    try {
      // config will be fetched from app-builder
      const widgetConfigs = await getPageWidget(pageCode, frameId, 'PROCESS_FORM');
      if (widgetConfigs.errors && widgetConfigs.errors.length) {
        throw widgetConfigs.errors[0];
      }
      const { config } = widgetConfigs.payload;
      const settings = (config.settings && JSON.parse(config.settings)) || {};

      const parsedSettings = Object.keys(settings).reduce(
        (acc, settingKey) => ({
          ...acc,
          [settingKey]: JSON.parse(settings[settingKey]),
        }),
        {}
      );

      return {
        ...config,
        settings: parsedSettings,
      };
    } catch (error) {
      this.handleError(error.message);
    }
    return null;
  }

  async fetchSchema() {
    const { config } = this.state;

    const connection = (config && config.knowledgeSource) || '';
    const processContainerId = (config && config.process) || '';

    try {
      const formSchema = await getProcessForm(connection, processContainerId);
      return formSchema;
    } catch (error) {
      this.handleError(error.message);
    }
    return null;
  }

  submitProcessForm(form) {
    this.setState({ submitting: true }, async () => {
      const { config } = this.state;
      const { onSubmitForm } = this.props;

      const connection = (config && config.knowledgeSource) || '';
      const processContainerId = (config && config.process) || '';

      try {
        const response = await postProcessForm(connection, processContainerId, form.formData);
        onSubmitForm({ ...form, response });
      } catch (error) {
        this.handleError(error.message);
      } finally {
        this.setState({ submitting: false });
      }
    });
  }

  handleError(errorMessage) {
    this.setState({ errorMessage });
    const { onError } = this.props;
    onError(errorMessage);
  }

  render() {
    const { loading, formSchema, config, submitting, errorMessage } = this.state;
    const { onError } = this.props;

    const uiSchema = (config && config.settings && config.settings.uiSchema) || {};

    return (
      <CustomEventContext.Provider
        value={{
          onSubmitForm: this.submitProcessForm,
          onError,
        }}
      >
        <ThemeProvider theme={theme}>
          <Container disableGutters>
            <WidgetBox>
              <JSONForm
                loading={loading}
                formSchema={formSchema}
                uiSchema={uiSchema}
                submitting={submitting}
              />
            </WidgetBox>
          </Container>
          <ErrorNotification message={errorMessage} onClose={this.closeNotification} />
        </ThemeProvider>
      </CustomEventContext.Provider>
    );
  }
}

ProcessFormContainer.propTypes = {
  onError: PropTypes.func,
  onSubmitForm: PropTypes.func,
  pageCode: PropTypes.string,
  frameId: PropTypes.string,
};

ProcessFormContainer.defaultProps = {
  onError: () => {},
  onSubmitForm: () => {},
  pageCode: '',
  frameId: '',
};

export default ProcessFormContainer;
