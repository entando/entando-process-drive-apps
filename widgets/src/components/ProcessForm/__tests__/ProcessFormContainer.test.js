import { render, wait } from '@testing-library/react';
import React from 'react';
import 'mocks/i18nMock';

import mockKeycloak from 'mocks/auth/keycloak';

import { DOMAINS } from 'api/constants';
import ProcessFormContainer from 'components/ProcessForm/ProcessFormContainer';
import WIDGETS from 'mocks/app-builder/widgets';
import WIDGET_CONFIGS from 'mocks/app-builder/pages';
import MOCKED_PROCESS_FORM from 'mocks/process-form/formSchema';

mockKeycloak();

describe('<ProcessFormContainer />', () => {
  it('renders snapshot correctly', async () => {
    const configUrl = `${DOMAINS.APP_BUILDER}/api/pages/${WIDGETS.PROCESS_FORM.pageCode}/widgets/${WIDGETS.PROCESS_FORM.frameId}`;
    const { config } = WIDGET_CONFIGS.PROCESS_FORM.payload;
    const processUrl = `${DOMAINS.PDA}/connections/${config.knowledgeSource}/processes/definitions/${config.process}/form`;

    fetch
      .once(JSON.stringify(WIDGET_CONFIGS.PROCESS_FORM))
      .once(JSON.stringify(MOCKED_PROCESS_FORM));

    const { container } = render(
      <ProcessFormContainer
        frameId={WIDGETS.PROCESS_FORM.frameId}
        pageCode={WIDGETS.PROCESS_FORM.pageCode}
      />
    );

    await wait(() => expect(fetch.mock.calls.length).toBe(2));

    expect(container).toMatchSnapshot();
    expect(fetch.mock.calls[0][0]).toEqual(configUrl);
    expect(fetch.mock.calls[1][0]).toEqual(processUrl);
  });

  it('renders snapshot correctly on error state', async () => {
    const { container } = render(<ProcessFormContainer config={{ knowledgeSource: 'test' }} />);

    await wait(() => expect(container).toMatchSnapshot());
  });
});
