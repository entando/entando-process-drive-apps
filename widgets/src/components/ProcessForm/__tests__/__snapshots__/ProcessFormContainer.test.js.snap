// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProcessFormContainer /> renders snapshot correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
  >
    <div
      class="MuiPaper-root WidgetBox-widgetBox-31 MuiPaper-outlined"
    >
      <div
        class="MuiBox-root MuiBox-root-61"
      >
        <div>
          <form
            class="JSONForm-themedForm-64"
          >
            <div
              class=""
            >
              <div>
                <div>
                  <div
                    class="makeStyles-header-95 makeStyles-header-100"
                  >
                    
                    
                  </div>
                  <div
                    class="MuiGrid-root-102 makeStyles-gridContainer-96 MuiGrid-container-103 MuiGrid-spacing-xs-3-127"
                  >
                    <div
                      class="MuiGrid-root-102 makeStyles-gridItem-99 makeStyles-gridItem-101 MuiGrid-item-104 MuiGrid-grid-xs-12-148"
                    >
                      <div
                        class=""
                      >
                        <div>
                          <div>
                            <div
                              class="makeStyles-header-95 makeStyles-header-205"
                            >
                              evaluation-taskform.frm
                              <hr
                                class="MuiDivider-root-207 makeStyles-divider-97"
                              />
                              
                            </div>
                            <div
                              class="MuiGrid-root-102 makeStyles-gridContainer-96 MuiGrid-container-103 MuiGrid-spacing-xs-3-127"
                            >
                              <div
                                class="MuiGrid-root-102 makeStyles-gridItem-99 makeStyles-gridItem-206 MuiGrid-item-104 MuiGrid-grid-xs-12-148"
                              >
                                <div
                                  class=""
                                >
                                  <label
                                    class="makeStyles-label-91"
                                    for="root_d1e6dd47-b24c-4f93-ba25-337832926124_employee"
                                  >
                                    Employee
                                     *
                                  </label>
                                  <div>
                                    <div
                                      class="MuiFormControl-root-214 MuiFormControl-fullWidth-217"
                                    >
                                      <div
                                        class="MuiFormControl-root-214 MuiTextField-root-218"
                                      >
                                        <label
                                          class="MuiFormLabel-root-231 MuiInputLabel-root-219 MuiInputLabel-formControl-225 MuiInputLabel-animated-228 MuiInputLabel-shrink-227 MuiInputLabel-outlined-230 Mui-focused Mui-focused Mui-required Mui-required"
                                          data-shrink="true"
                                          for="root_d1e6dd47-b24c-4f93-ba25-337832926124_employee"
                                          id="root_d1e6dd47-b24c-4f93-ba25-337832926124_employee-label"
                                        >
                                          Employee
                                          <span
                                            class="MuiFormLabel-asterisk-238 MuiInputLabel-asterisk-224"
                                          >
                                             
                                            *
                                          </span>
                                        </label>
                                        <div
                                          class="MuiInputBase-root-254 MuiOutlinedInput-root-239 Mui-focused Mui-focused MuiInputBase-formControl-255"
                                        >
                                          <input
                                            aria-invalid="false"
                                            class="MuiInputBase-input-265 MuiOutlinedInput-input-249 MuiInputBase-inputHiddenLabel-271"
                                            id="root_d1e6dd47-b24c-4f93-ba25-337832926124_employee"
                                            required=""
                                            type="string"
                                            value=""
                                          />
                                          <fieldset
                                            aria-hidden="true"
                                            class="PrivateNotchedOutline-root-272 MuiOutlinedInput-notchedOutline-248"
                                          >
                                            <legend
                                              class="PrivateNotchedOutline-legendLabelled-274"
                                            >
                                              <span>
                                                Employee
                                                 *
                                              </span>
                                            </legend>
                                          </fieldset>
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                              <div
                                class="MuiGrid-root-102 makeStyles-gridItem-99 makeStyles-gridItem-206 MuiGrid-item-104 MuiGrid-grid-xs-12-148"
                              >
                                <div
                                  class=""
                                >
                                  <label
                                    class="makeStyles-label-91"
                                    for="root_d1e6dd47-b24c-4f93-ba25-337832926124_reason"
                                  >
                                    Reason
                                     *
                                  </label>
                                  <div>
                                    <div
                                      class="MuiFormControl-root-214 MuiFormControl-fullWidth-217"
                                    >
                                      <div
                                        class="MuiFormControl-root-214 MuiTextField-root-218"
                                      >
                                        <label
                                          class="MuiFormLabel-root-231 MuiInputLabel-root-219 MuiInputLabel-formControl-225 MuiInputLabel-animated-228 MuiInputLabel-shrink-227 MuiInputLabel-outlined-230 Mui-required Mui-required"
                                          data-shrink="true"
                                          for="root_d1e6dd47-b24c-4f93-ba25-337832926124_reason"
                                          id="root_d1e6dd47-b24c-4f93-ba25-337832926124_reason-label"
                                        >
                                          Reason
                                          <span
                                            class="MuiFormLabel-asterisk-238 MuiInputLabel-asterisk-224"
                                          >
                                             
                                            *
                                          </span>
                                        </label>
                                        <div
                                          class="MuiInputBase-root-254 MuiOutlinedInput-root-239 MuiInputBase-formControl-255 MuiInputBase-multiline-262 MuiOutlinedInput-multiline-247"
                                        >
                                          <textarea
                                            aria-invalid="false"
                                            class="MuiInputBase-input-265 MuiOutlinedInput-input-249 MuiInputBase-inputMultiline-267 MuiOutlinedInput-inputMultiline-251 MuiInputBase-inputHiddenLabel-271"
                                            id="root_d1e6dd47-b24c-4f93-ba25-337832926124_reason"
                                            placeholder=""
                                            required=""
                                            rows="5"
                                          />
                                          <fieldset
                                            aria-hidden="true"
                                            class="PrivateNotchedOutline-root-272 MuiOutlinedInput-notchedOutline-248"
                                          >
                                            <legend
                                              class="PrivateNotchedOutline-legendLabelled-274"
                                            >
                                              <span>
                                                Reason
                                                 *
                                              </span>
                                            </legend>
                                          </fieldset>
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                              <div
                                class="MuiGrid-root-102 makeStyles-gridItem-99 makeStyles-gridItem-206 MuiGrid-item-104 MuiGrid-grid-xs-12-148"
                              >
                                <div
                                  class=""
                                >
                                  <label
                                    class="makeStyles-label-91"
                                    for="root_d1e6dd47-b24c-4f93-ba25-337832926124_list"
                                  >
                                    List
                                     *
                                  </label>
                                  <div>
                                    <div
                                      class="MuiFormControl-root-214 MuiFormControl-fullWidth-217"
                                    >
                                      <div
                                        class="MuiFormControl-root-214 MuiTextField-root-218"
                                      >
                                        <label
                                          class="MuiFormLabel-root-231 MuiInputLabel-root-219 MuiInputLabel-formControl-225 MuiInputLabel-animated-228 MuiInputLabel-shrink-227 MuiInputLabel-outlined-230 Mui-required Mui-required"
                                          data-shrink="true"
                                          for="root_d1e6dd47-b24c-4f93-ba25-337832926124_list"
                                          id="root_d1e6dd47-b24c-4f93-ba25-337832926124_list-label"
                                        >
                                          List
                                          <span
                                            class="MuiFormLabel-asterisk-238 MuiInputLabel-asterisk-224"
                                          >
                                             
                                            *
                                          </span>
                                        </label>
                                        <div
                                          class="MuiInputBase-root-254 MuiOutlinedInput-root-239 MuiInputBase-formControl-255"
                                        >
                                          <input
                                            aria-invalid="false"
                                            class="MuiInputBase-input-265 MuiOutlinedInput-input-249 MuiInputBase-inputHiddenLabel-271"
                                            id="root_d1e6dd47-b24c-4f93-ba25-337832926124_list"
                                            required=""
                                            type="string"
                                            value=""
                                          />
                                          <fieldset
                                            aria-hidden="true"
                                            class="PrivateNotchedOutline-root-272 MuiOutlinedInput-notchedOutline-248"
                                          >
                                            <legend
                                              class="PrivateNotchedOutline-legendLabelled-274"
                                            >
                                              <span>
                                                List
                                                 *
                                              </span>
                                            </legend>
                                          </fieldset>
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                              <div
                                class="MuiGrid-root-102 makeStyles-gridItem-99 makeStyles-gridItem-206 MuiGrid-item-104 MuiGrid-grid-xs-12-148"
                              >
                                <div
                                  class=""
                                >
                                  <div>
                                    <div
                                      class="MuiFormControl-root-214 MuiFormControl-fullWidth-217"
                                    >
                                      <label
                                        class="MuiFormControlLabel-root-276"
                                      >
                                        <span
                                          aria-disabled="false"
                                          class="MuiButtonBase-root-301 MuiIconButton-root-292 PrivateSwitchBase-root-288 MuiCheckbox-root-282 MuiCheckbox-colorPrimary-286 MuiIconButton-colorPrimary-296"
                                        >
                                          <span
                                            class="MuiIconButton-label-300"
                                          >
                                            <input
                                              class="PrivateSwitchBase-input-291"
                                              data-indeterminate="false"
                                              id="root_d1e6dd47-b24c-4f93-ba25-337832926124_checkBox"
                                              type="checkbox"
                                              value=""
                                            />
                                            <svg
                                              aria-hidden="true"
                                              class="MuiSvgIcon-root-304"
                                              focusable="false"
                                              role="presentation"
                                              viewBox="0 0 24 24"
                                            >
                                              <path
                                                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                                              />
                                            </svg>
                                          </span>
                                          <span
                                            class="MuiTouchRipple-root-372"
                                          />
                                        </span>
                                        <span
                                          class="MuiTypography-root-313 MuiFormControlLabel-label-281 MuiTypography-body1-315"
                                        >
                                          Agree
                                        </span>
                                      </label>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="JSONForm-actionButtons-62"
            >
              <button
                class="MuiButtonBase-root-301 MuiButton-root-343 MuiButton-contained-351 MuiButton-containedPrimary-352"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label-344"
                >
                  common.submit
                </span>
                <span
                  class="MuiTouchRipple-root-372"
                />
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  
</div>
`;

exports[`<ProcessFormContainer /> renders snapshot correctly on error state 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
  >
    <div
      class="MuiPaper-root WidgetBox-widgetBox-412 MuiPaper-outlined"
    >
      <div
        class="MuiBox-root MuiBox-root-442"
      >
        messages.warnings.noFormSchema
      </div>
    </div>
  </div>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
  >
    <div
      class="MuiPaper-root MuiSnackbarContent-root Notification-error-449 MuiPaper-elevation6"
      direction="up"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message"
      >
        <span
          class="Notification-message-446"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root Notification-icon-447 Notification-iconVariant-448"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
          invalid json response body at undefined reason: Unexpected end of JSON input
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action"
      >
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root Notification-icon-447"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
